jiant/preprocess.py

```diff
@@ -318,28 +323,66 @@ def build_tasks(args):
             word_embs = pkl.load(open(emb_file, "rb"))
         log.info("Trimmed word embeddings: %s", str(word_embs.size()))
 
-    # 4) Index tasks using vocab (if preprocessed copy not available).
-    preproc_dir = os.path.join(args.exp_dir, "preproc")
-    utils.maybe_make_dir(preproc_dir)
-    reindex_tasks = parse_task_list_arg(args.reindex_tasks)
-    utils.assert_for_log(
-        not (args.reload_indexing and not reindex_tasks),
-        'Flag reload_indexing was set, but no tasks are set to reindex (use -o "args.reindex_tasks'
-        ' = "task1,task2,..."")',
-    )
+    # 4) Set up task modulator, this includes boundary function and model flags
+    boundary_token_fn = None
+    lm_boundary_token_fn = None
 
-    # Set up boundary_token_fn, which applies SOS/EOS/SEP/CLS delimiters
-    if args.input_module.startswith("bert"):
+    if args.input_module.startswith("bert-"):
         from jiant.pytorch_transformers_interface.modules import BertEmbedderModule
 
         boundary_token_fn = BertEmbedderModule.apply_boundary_tokens
-    elif args.input_module.startswith("xlnet"):
+    elif args.input_module.startswith("roberta-"):
+        from jiant.pytorch_transformers_interface.modules import RobertaEmbedderModule
+
+        boundary_token_fn = RobertaEmbedderModule.apply_boundary_tokens
+    elif args.input_module.startswith("xlnet-"):
         from jiant.pytorch_transformers_interface.modules import XLNetEmbedderModule
 
         boundary_token_fn = XLNetEmbedderModule.apply_boundary_tokens
+    elif args.input_module.startswith("openai-gpt"):
+        from jiant.pytorch_transformers_interface.modules import OpenAIGPTEmbedderModule
+
+        boundary_token_fn = OpenAIGPTEmbedderModule.apply_boundary_tokens
+        lm_boundary_token_fn = OpenAIGPTEmbedderModule.apply_lm_boundary_tokens
+    elif args.input_module.startswith("gpt2"):
+        from jiant.pytorch_transformers_interface.modules import GPT2EmbedderModule
+
+        boundary_token_fn = GPT2EmbedderModule.apply_boundary_tokens
+        lm_boundary_token_fn = GPT2EmbedderModule.apply_lm_boundary_tokens
+    elif args.input_module.startswith("transfo-xl-"):
+        from jiant.pytorch_transformers_interface.modules import TransfoXLEmbedderModule
+
+        boundary_token_fn = TransfoXLEmbedderModule.apply_boundary_tokens
+        lm_boundary_token_fn = TransfoXLEmbedderModule.apply_lm_boundary_tokens
+    elif args.input_module.startswith("xlm-"):
+        from jiant.pytorch_transformers_interface.modules import XLMEmbedderModule
+
+        boundary_token_fn = XLMEmbedderModule.apply_boundary_tokens
     else:
         boundary_token_fn = utils.apply_standard_boundary_tokens
 
+    from jiant.models import input_module_uses_pair_embedding, input_module_uses_mirrored_pair
```

Comment:
Move this up to the rest of the imports

Reply:
This can not be done. preprocess.py import model.py, model.py import pytorch_transformer_interface, pytorch_transformer_interce import parse_task_list_arg from preprocess.py, then we get a cycle import. Surly this import here in preprocess is ugly, I think the most reasonable way to break the chain is moving parse_task_list_arg to utils or replace the string-form args.target_tasks with parsed output from parse_task_list_arg immediately when the program starts. But I don't want to make too many peripheral changes in one PR, so I used this ugly import.